resize.c

```diff
@@ -42,50 +43,95 @@
 #endif
 #endif
 
-/**
- * mutt_resize_screen - Called after receiving SIGWINCH
- */
-void mutt_resize_screen(void)
+int mutt_resize_getrows(int rows)
 {
   const char *cp = NULL;
-  int fd;
-  struct winsize w;
-#if !defined(USE_SLANG_CURSES)
-  int SLtt_Screen_Rows, SLtt_Screen_Cols;
-#endif
+  cp = mutt_str_getenv("LINES");
+  if (cp && (mutt_str_atoi(cp, &rows) < 0))
+    return 24;
+  return -1;
+}
 
-  SLtt_Screen_Rows = -1;
-  SLtt_Screen_Cols = -1;
+int mutt_resize_getcols(int cols)
+{
+  const char *cp = NULL;
+  cp = mutt_str_getenv("COLUMNS");
+  if (cp && (mutt_str_atoi(cp, &cols) < 0))
+    return 80;
+  return -1;
+}
+
+int mutt_resize_getwinsize(winsize w)
```

Comment:
You're missing a `struct`

Reply:
This needs to be a pointer, otherwise we're working with a local copy of the struct